src/java.base/share/classes/java/lang/constant/MethodTypeDesc.java

```diff
@@ -55,6 +56,36 @@ static MethodTypeDesc ofDescriptor(String descriptor) {
         return MethodTypeDescImpl.ofDescriptor(descriptor);
     }
 
+    /**
+     * Returns a {@linkplain MethodTypeDesc} with the given return type and no
+     * parameter types.
+     *
+     * @param returnDesc a {@linkplain ClassDesc} describing the return type
+     * @return a {@linkplain MethodTypeDesc} describing the desired method type
+     * @throws NullPointerException if {@code returnDesc} is {@code null}
+     * @since 21
+     */
+    static MethodTypeDesc of(ClassDesc returnDesc) {
+        return new MethodTypeDescImpl(returnDesc, ConstantUtils.EMPTY_CLASSDESC);
+    }
+
+    /**
+     * Returns a {@linkplain MethodTypeDesc} given the return type and parameter
+     * types.
+     *
+     * @param returnDesc a {@linkplain ClassDesc} describing the return type
+     * @param paramDescs a {@linkplain Collection} of {@linkplain ClassDesc}s
+     * describing the argument types, in its iteration order
+     * @return a {@linkplain MethodTypeDesc} describing the desired method type
+     * @throws NullPointerException if any argument or its contents are {@code null}
+     * @throws IllegalArgumentException if any element of {@code paramDescs} is a
+     * {@link ClassDesc} for {@code void}
+     * @since 21
+     */
+    static MethodTypeDesc of(ClassDesc returnDesc, Collection<ClassDesc> paramDescs) {
```

Comment:
This needs to be:
```suggestion
    static MethodTypeDesc of(ClassDesc returnDesc, List<ClassDesc> paramDescs) {
```

as `Collection` doesn’t have a defined encounter order.

Reply:
No, this is what List.copyOf supports.

Reply:
Alternatively, we can switch to SequencedCollection when it's integrated.

Reply:
[`MethodType.methodType(…)`] also doesn’t allow plain `Collection`s for the same reason, even though `List.copyOf` supports creation from a `Collection`.

[`MethodType.methodType(…)`]: https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/lang/invoke/MethodType.html#methodType(java.lang.Class,java.util.List)

Reply:
After thinking, users who want to pass in a `Collection` can just do `MethodTypeDesc.of(returnDesc, List.copyOf(paramDescCollection))` so using a `List` is good.